sky/global_user_state.py

```diff
@@ -390,6 +393,32 @@ def set_cluster_metadata(cluster_name: str, metadata: Dict[str, Any]) -> None:
         raise ValueError(f'Cluster {cluster_name} not found.')
 
 
+def get_cluster_storage_mounts_metadata(
+        cluster_name: str) -> Optional[Dict[str, Any]]:
+    rows = _DB.cursor.execute(
+        'SELECT storage_mounts_metadata FROM clusters WHERE name=(?)',
+        (cluster_name,))
+    for (storage_mounts_metadata,) in rows:
+        if storage_mounts_metadata is None:
+            return None
+        return pickle.loads(storage_mounts_metadata)
+    return None
+
+
+def set_cluster_storage_mounts_metadata(
+        cluster_name: str, storage_mounts_metadata: Dict[str, Any]) -> None:
+    _DB.cursor.execute(
+        'UPDATE clusters SET storage_mounts_metadata=(?) WHERE name=(?)', (
+            pickle.dumps(storage_mounts_metadata),
```

Comment:
We don't have a lock for the cluster when doing file mounts, which means if there are two `sky launch` on the same  cluster, there can be a race condition writing into the storage metadata column. 

Do we need to adopt some file lock and do a read, update, dump?
```
with filelock.FileLock(...):
    cluster_metadata = get_cluster_storage_mounts_metadata(cluster_name)
    cluster_metadata.update(storage_mounts_metadata)
    _DB.cursor.execute(...)
```